### PR TITLE
fix(lsp): normalize case sensitive `root_dir` on Windows

### PR DESCRIPTION
**Pull request by TrungNguyen153**

### CODE REVIEW

File: lua/rustaceanvim/lsp.lua [hunk 1]
```diff
@@ -105,6 +105,14 @@ M.start = function(bufnr)
   ---@type RustaceanLspClientConfig
   local lsp_start_opts = vim.tbl_deep_extend('force', {}, client_config)
   local root_dir = get_root_dir(vim.api.nvim_buf_get_name(bufnr))
+   local function normalize_path(path)
```
Comment (mrcjkb) [inline lua/rustaceanvim/lsp.lua:108]: - Let's move this function to the top level (and give it `---@param` / `---@return` type annotations).
- You will need to run stylua (which is why CI is currently failing).

**Review by mrcjkb — CHANGES_REQUESTED** [review]
Thanks for the PR!

File: lua/rustaceanvim/lsp.lua (hunk 1 above)
Comment (TrungNguyen153) [reply lua/rustaceanvim/lsp.lua]: kk let me try. My first pull request.

File: lua/rustaceanvim/lsp.lua [hunk 2]
```diff
@@ -96,6 +96,17 @@ local function is_in_workspace(client, root_dir)
   return false
 end
 
+--- Fast fix for root_dir fix on windows
+---@param path string
+---@return string normalize_path
+local function normalize_path(path)
+  local has_windows_drive_letter = path:match '^%a:'
```
Comment (mrcjkb) [inline lua/rustaceanvim/lsp.lua:103]: Sorry, it looks like my other comment got deleted.

I'm not sure about this, but I think we might want to check if we're on Windows here, with `require('rustaceanvim.shell').is_windows()`

Comment (TrungNguyen153) [reply lua/rustaceanvim/lsp.lua:103]: kk i will use this for avoid unexpected behavior

**Comment by mrcjkb** [comment]
Looks good. Only the stylua check is failing now.

File: lua/rustaceanvim/lsp.lua [hunk 3]
```diff
@@ -96,6 +96,19 @@ local function is_in_workspace(client, root_dir)
...
+---@return string normalize_path
+local function normalize_path(path)
+  if require('rustaceanvim.shell').is_windows() then
+    local has_windows_drive_letter = path:match '^%a:'
+    if has_windows_drive_letter then
+      return path:sub(1,1):lower()..path:sub(2)
+    end
+  end
+  return path
+end
```
Comment (mrcjkb) [inline lua/rustaceanvim/lsp.lua:110]:
```suggestion
---Normalize path for Windows, which is case insensitive
---@param path string
---@return string normalize_path
local function normalize_path(path)
  if require('rustaceanvim.shell').is_windows() then
    local has_windows_drive_letter = path:match('^%a:')
    if has_windows_drive_letter then
      return path:sub(1, 1):lower() .. path:sub(2)
    end
  end
  return path
end
```

**Review by mrcjkb — APPROVED** [review]

File: lua/rustaceanvim/lsp.lua [hunk 4]
```diff
@@ -105,6 +118,7 @@ M.start = function(bufnr)
   ---@type RustaceanLspClientConfig
   local lsp_start_opts = vim.tbl_deep_extend('force', {}, client_config)
   local root_dir = get_root_dir(vim.api.nvim_buf_get_name(bufnr))
+  root_dir = normalize_path(root_dir)
```
Comment (mrcjkb) [inline lua/rustaceanvim/lsp.lua:121]:
```suggestion
  root_dir = root_dir and normalize_path(root_dir)
```

**Review by mrcjkb — DISMISSED** [review]

**Comment by TrungNguyen153** [comment]
Tks for merge it. Work as expected. Sorry about not response.

**Pull request merged**